doc/user-guide/data_model.rst

```diff
@@ -647,47 +651,48 @@ cannot be made so because field data is not expected to match the
 number of cells or points.  As such, it also cannot be plotted.
 
 
-Vectors, Textures, and Normals Attributes
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Both cell and point data can also store the following "special" attributes in addition to :attr:`active_scalars <pyvista.core.dataset.DataSet.active_scalars>`
+Vectors, Texture Coords, and Normals Attributes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Both cell and point data can also store the following "special" attributes in addition to :attr:`active_scalars <pyvista.DataSet.active_scalars>`:
 
-* :attr:`active_normals <pyvista.core.dataset.DataSet.active_normals>`
-* :attr:`active_t_coords <pyvista.core.dataset.DataSet.active_t_coords>`
-* :attr:`active_vectors <pyvista.core.dataset.DataSet.active_vectors>`
+* :attr:`active_normals <pyvista.DataSet.active_normals>`
+* :attr:`active_t_coords <pyvista.DataSet.active_t_coords>`
+* :attr:`active_vectors <pyvista.DataSet.active_vectors>`
 
 
 Active Normals
 ~~~~~~~~~~~~~~
 The :attr:`active_normals
-<pyvista.core.dataset.DataSet.active_normals>` array is a special
-array that is used for creating physically based rendering or
-rendering smooth shading using the phong interpolation.  If this array
+<pyvista.DataSet.active_normals>` array is a special array that
+specifies the local normal direction of meshes. It is used for
+creating physically based rendering, rendering smooth shading using
+Phong interpolation, warping by scalars, etc.  If this array
```

Comment:
I'm pretty sure warping also needs normals. More importantly. from a quick check it seemed to me that when normals are not available on a mesh, warping can go haywire. Have to check if we're computing normals when they are missing.

Reply:
Good point.  Let's open a different issue for this.

Reply:
Done in https://github.com/pyvista/pyvista/issues/1644